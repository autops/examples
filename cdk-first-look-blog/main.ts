import { Construct } from 'constructs';
import { App, Chart, ChartProps, Size, JsonPatch, ApiObject } from 'cdk8s';
import {Cpu, Deployment, DeploymentStrategy, EnvValue, Ingress, IngressBackend, PercentOrAbsolute} from 'cdk8s-plus-22';

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = { }) {
    super(scope, id, props);

    const deployment = new Deployment(this, 'deployment', {
      metadata:{
        annotations: {
          'prometheus.io/scrape': "true",
          'prometheus.io/port': "9797",
        },
        labels: {
          app: 'podinfo',
          team: 'cest',
        }
      },
      strategy: DeploymentStrategy.rollingUpdate({maxUnavailable: PercentOrAbsolute.absolute(0)}),
      containers:[{
        image: 'ghcr.io/stefanprodan/podinfo:6.2.2',
        args: [
          './podinfo',
          '--port=9898',
          '--port-metrics=9797',
          '--grpc-port=9999',
          '--grpc-service-name=podinfo',
          '--level=debug',
          '--random-delay=false',
          '--random-error=false',
        ],
        envVariables: {
          PODINFO_UI_COLOR: EnvValue.fromValue('#34577c')
        },
        resources: {
          cpu: {
            limit: Cpu.millis(800),
            request: Cpu.millis(100),
          },
          memory: {
            limit: Size.mebibytes(500),
            request: Size.mebibytes(100),
          }
        },
        ports: [
          {
            number: 9898,
            name: 'http',
          },
          {
            number: 9797,
            name: 'http-metrics',
          },
          {
            number: 9999,
            name: 'grpc',
          }
        ]
      }]
    })
    
    const backend = IngressBackend.fromService(deployment.exposeViaService({ports: [{port: 9898}]}));
    
    const ingress = new Ingress(this, 'ingress', {
      metadata: {
        annotations: {
          'alb.ingress.kubernetes.io/scheme': 'internet-facing',
          'alb.ingress.kubernetes.io/target-type': 'ip',
          'alb.ingress.kubernetes.io/subnets': 'subnet-004678903456785,subnet-0b1321321321e212,subnet-06489d92e5839353',
          'alb.ingress.kubernetes.io/tags': 'Application=podinfo, Automation=kubernetes',
          'alb.ingress.kubernetes.io/certificate-arn': 'arn:aws:acm:eu-west-1:09876543211:certificate/56fff548-218d-4f0e-9965-54548697784',
        },
      }
    });
    ingress.addHostRule('k8s-podinfo.mydomain.tld', '/', backend)
    ingress.addTls([{hosts: ['k8s-podinfo.mydomain.tld']}])
   
    // Add ingressClassName (not supported by CDK atm)
    const ing = ApiObject.of(ingress)
    ing?.addJsonPatch(JsonPatch.add('/spec/ingressClassName', 'alb'))

  }
}

const app = new App();
new MyChart(app, 'cdk');
app.synth();
