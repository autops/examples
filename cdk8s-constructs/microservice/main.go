package main

import (
	"github.com/autops/examples/microservice/lib/microservice"

	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
	"github.com/cdk8s-team/cdk8s-core-go/cdk8s/v2"
)

type MyChartProps struct {
	cdk8s.ChartProps
}

func NewMyChart(scope constructs.Construct, id string, props *MyChartProps) cdk8s.Chart {
	var cprops cdk8s.ChartProps
	if props != nil {
		cprops = props.ChartProps
	}

	chart := cdk8s.NewChart(scope, jsii.String(id), &cprops)

	// Create new Microservice
	ms := microservice.New(chart, jsii.String("ms"), microservice.ContextProps{
		Name: jsii.String("potato"),
	})

	// Add Vault config (annotations) to the Microservice passing the name of the Role in Vault
	ms.AddVaultRole("potato-role")

	return chart
}

func main() {
	app := cdk8s.NewApp(nil)
	NewMyChart(app, "potato-ms", nil)
	app.Synth()
}
