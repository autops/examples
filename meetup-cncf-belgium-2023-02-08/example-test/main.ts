import { Construct } from 'constructs';
import { App, Chart } from 'cdk8s';
import { Deployment, Ingress, IngressBackend, } from 'cdk8s-plus-25';

export interface MyAppProps {
  readonly env?: string;
}

export class MyApp extends Chart {
  constructor(scope: Construct, id: string, props: MyAppProps) {
    super(scope, id);

    // define resources here
    const depl = new Deployment(this, "AppDeploy", {
      containers: [
        {
          image: 'nginx',
          ports: [{
            number: 9000
          }]
        }
      ],
      metadata: {
        labels: {
          env: props.env || 'dev'
        }
      }
    })

    //depl.exposeViaService({ports: [{port: 9000}]})
    //
    const ingress = new Ingress(this, 'AppIngres', {})
    const backend = IngressBackend.fromService(depl.exposeViaService({ports: [{port: 9000}]}));
    ingress.addHostRule('myapp.autops.cloud', '/', backend )

  }


}

const app = new App();
new MyApp(app, 'production', {env: 'production'});
new MyApp(app, 'development', {env: 'development'});
app.synth();
