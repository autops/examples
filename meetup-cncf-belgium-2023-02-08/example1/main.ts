import { Construct } from 'constructs';
import { App, Chart} from 'cdk8s';
import { Deployment, Ingress, IngressBackend } from 'cdk8s-plus-25';

export interface Myprops {
  readonly env?: string;
}

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: Myprops) {
    super(scope, id);

    // define resources here
    const depl = new Deployment(this, 'Deploy', {
      containers: [
        {
          image: 'nginx',
          ports: [{
            number: 9000
          }]
        }
      ],
      metadata: {
        labels: {
          env: props.env || 'development'
        }
      }
    })

    const ingress = new Ingress(this, 'AppIngres', {})
    const backend = IngressBackend.fromService(depl.exposeViaService({ports: [{port: 9000}]}));
    ingress.addHostRule('myapp.autops.cloud', '/', backend )

  }
}

const app = new App();
new MyChart(app, 'production', {env: 'production'});
new MyChart(app, 'development', {env: 'development'});
app.synth();
