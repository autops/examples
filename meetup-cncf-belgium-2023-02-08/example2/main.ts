import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';
import {CompanyService} from './lib/company-service'

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = { }) {
    super(scope, id, props);

    // define resources here
    new CompanyService(this, 'NewService', {
      name: 'foo',
      image: 'nginx'
    })

  }
}

const app = new App();
new MyChart(app, 'example2');
app.synth();
