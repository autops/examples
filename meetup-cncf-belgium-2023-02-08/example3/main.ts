import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';
import * as crd from './imports/cert-manager.io'

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = { }) {
    super(scope, id, props);

    // define resources here
    new crd.Certificate(this, 'Cert', {
      spec: {
        secretName: 'mysecret',
        issuerRef: {
          name: 'mysecretissuer'
        }
      }
    })

  }
}

const app = new App();
new MyChart(app, 'example3');
app.synth();
