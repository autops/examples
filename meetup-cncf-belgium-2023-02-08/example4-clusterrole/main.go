package main

import (
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
	"github.com/cdk8s-team/cdk8s-core-go/cdk8s/v2"
	cdk8splus "github.com/cdk8s-team/cdk8s-plus-go/cdk8splus27/v2"
)

type MyChartProps struct {
	cdk8s.ChartProps
}

func NewMyChart(scope constructs.Construct, id string, props *MyChartProps) cdk8s.Chart {
	var cprops cdk8s.ChartProps
	if props != nil {
		cprops = props.ChartProps
	}
	chart := cdk8s.NewChart(scope, jsii.String(id), &cprops)

	eventsAPI := cdk8splus.ApiResource_PODS()

	cdk8splus.NewClusterRole(chart, jsii.String("flux-pubsub-publisher-monitor-kustomizations"), &cdk8splus.ClusterRoleProps{
		Metadata: &cdk8s.ApiObjectMetadata{
			Namespace: jsii.String("flux-system"),
		},
		Rules: &[]*cdk8splus.ClusterRolePolicyRule{
			{
				Endpoints: &[]cdk8splus.IApiEndpoint{eventsAPI},
				Verbs: &[]*string{
					jsii.String("get"),
					jsii.String("list"),
					jsii.String("watch"),
				},
			},
		},
	})

	return chart
}

func main() {
	app := cdk8s.NewApp(nil)
	NewMyChart(app, "example4-clusterrole", nil)
	app.Synth()
}
