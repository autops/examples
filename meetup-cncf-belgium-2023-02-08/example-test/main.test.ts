import {MyApp} from './main';
import {Testing} from 'cdk8s';

describe('Placeholder', () => {
  test('Empty', () => {
    const app = Testing.app();
    const chart = new MyApp(app, 'test-chart', {});
    const results = Testing.synth(chart)
    //console.log(results)
    expect(results).toMatchSnapshot();
  });

  test('Label shoud match', () => {
    const app = Testing.app();
    const chart = new MyApp(app, 'test-chart', {env: 'production'});
    const results = Testing.synth(chart)
    expect(results[0]).toHaveProperty('metadata.labels.env', 'production')
  })

  test('Production needs 2 replicas', () => {
    const app = Testing.app();
    const chart = new MyApp(app, 'test-chart', {env: 'production'});
    const results = Testing.synth(chart)
    //expect(results[0]).toHaveProperty('spec.replicas', 2)
    const deploy = results.filter(v => { return v.kind == 'Deployment'})
    expect(deploy[0]).toHaveProperty('spec.replicas', 2);
  })
});
