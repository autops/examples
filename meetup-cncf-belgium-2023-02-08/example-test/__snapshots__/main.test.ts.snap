// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "env": "dev",
      },
      "name": "test-chart-appdeploy-c8a58070",
    },
    "spec": Object {
      "minReadySeconds": 0,
      "progressDeadlineSeconds": 600,
      "replicas": 2,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.io/metadata.addr": "test-chart-AppDeploy-c865d7ea",
        },
      },
      "strategy": Object {
        "rollingUpdate": Object {
          "maxSurge": "25%",
          "maxUnavailable": "25%",
        },
        "type": "RollingUpdate",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.io/metadata.addr": "test-chart-AppDeploy-c865d7ea",
          },
        },
        "spec": Object {
          "automountServiceAccountToken": false,
          "containers": Array [
            Object {
              "image": "nginx",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": Array [
                Object {
                  "containerPort": 9000,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "1500m",
                  "memory": "2048Mi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "512Mi",
                },
              },
              "securityContext": Object {
                "allowPrivilegeEscalation": false,
                "privileged": false,
                "readOnlyRootFilesystem": true,
                "runAsNonRoot": true,
              },
            },
          ],
          "dnsPolicy": "ClusterFirst",
          "restartPolicy": "Always",
          "securityContext": Object {
            "fsGroupChangePolicy": "Always",
            "runAsNonRoot": true,
          },
          "setHostnameAsFQDN": false,
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-chart-appdeploy-service-c844bcf4",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 9000,
        },
      ],
      "selector": Object {
        "cdk8s.io/metadata.addr": "test-chart-AppDeploy-c865d7ea",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1",
    "kind": "Ingress",
    "metadata": Object {
      "name": "test-chart-appingres-c8316167",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "myapp.autops.cloud",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "service": Object {
                    "name": "test-chart-appdeploy-service-c844bcf4",
                    "port": Object {
                      "number": 9000,
                    },
                  },
                },
                "path": "/",
                "pathType": "Prefix",
              },
            ],
          },
        },
      ],
    },
  },
]
`;
